src/main/java/hudson/plugins/git/extensions/impl/FirstBuildChangelog.java

```diff
@@ -0,0 +1,74 @@
+package hudson.plugins.git.extensions.impl;
+
+import hudson.Extension;
+import hudson.plugins.git.extensions.GitSCMExtension;
+import hudson.plugins.git.extensions.GitSCMExtensionDescriptor;
+import java.util.Objects;
+import org.jenkinsci.Symbol;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+
+/**
+ * First Build generates a changelog.
+ *
+ * @author Derek Inskeep
+ */
+public class FirstBuildChangelog extends GitSCMExtension {
+    private boolean makeChangelog;
+
+    @DataBoundConstructor
+    public FirstBuildChangelog() {}
+
+    public boolean isMakeChangelog() {
+        return makeChangelog;
+    }
+
+    @DataBoundSetter
+    public void setMakeChangelog(boolean makeChangelog) {
+        this.makeChangelog = makeChangelog;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        FirstBuildChangelog that = (FirstBuildChangelog) o;
+        return makeChangelog == that.makeChangelog;
```

Comment:
This is unexpected (at least by me) in an equals method.   I'm not aware of any other equals method in the plugin that uses an exact equality test in the `equals` method.  Please use equalsverifier to confirm that the equals contract and the hashCode contract are fulfilled by this code.

I'm not sure that my suggestion is the correct implementation of equals either, but equalsverifier will allow you to declare your expectations for equality in the test.
```suggestion
        return Objects.equals(makeChangelog, that.makeChangelog);
```

Reply:
My sincere apologies for being a terrible reviewer of code.  The exact equals comparison that was originally here is exactly the correct thing to be because `makeChangelog` is a primitive type (a boolean).  I should have looked at the type of the field before making my comment!